pkg/api/routes.go

```diff
@@ -98,15 +98,15 @@ func (rh *RouteHandler) SetupRoutes() {
 		g.HandleFunc(fmt.Sprintf("/{name:%s}/blobs/{digest}", NameRegexp.String()),
 			rh.blobLockWrapper(rh.DeleteBlob)).Methods("DELETE")
 		g.HandleFunc(fmt.Sprintf("/{name:%s}/blobs/uploads/", NameRegexp.String()),
-			rh.blobLockWrapper(rh.CreateBlobUpload)).Methods("POST")
```

Comment:
Locks are removed from CreateBlobUpload, PatchBlobUpload, UpdateBlobUpload, all this handler calls https://github.com/anuvu/zot/blob/ea0ef61a654b47e06ec978a53f176419dd5f0277/pkg/storage/storage.go#L111 internally and this method initializes repository, may be we need a lock here also (inside InitRepo method).

Reply:
> Locks are removed from CreateBlobUpload, PatchBlobUpload, UpdateBlobUpload, all this handler calls 
Not removed, they have been moved into pkg/storage/storage.go

Reply:
Yes but InitRepo is called everywhere before lock is acquired, I think we also need a lock there.